test/components/guideLineLayerTests.ts

```diff
@@ -1,481 +1,475 @@
+        assert.strictEqual(gll.value(), setValue, "value is not changed");
+        assert.strictEqual((<any>linearScale)._callbacks.size, 0, "callback was removed from the previous Scale");
+      });
+
+      it("updates value() when changing the scale if pixelPosition() was the last property set", () => {
+        let setValue = 0.5;
+        gll.value(setValue);
+        let setPosition = -100;
+        gll.pixelPosition(setPosition);
+        let expectedValue = linearScale.invert(setPosition);
+
+        gll.scale(linearScale);
+        assert.strictEqual(gll.value(), expectedValue,
+          "setting the scale updates the value if pixelPosition() was the last thing set");
+        assert.strictEqual(gll.pixelPosition(), setPosition, "pixel position is not changed");
+        assert.notStrictEqual(gll.value(), setValue, "originally-set value was overridden");
+
+        let linearScaleB = new Plottable.Scales.Linear();
+        linearScaleB.domain([0, 1]);
+        linearScaleB.range([0, 200]);
+        let expectedValueB = linearScaleB.invert(setPosition);
+        gll.scale(linearScaleB);
+        assert.strictEqual(gll.value(), expectedValueB,
+          "changing the scale updates the value if pixelPosition() was the last thing set");
+        assert.strictEqual(gll.pixelPosition(), setPosition, "pixel position is not changed");
+        assert.strictEqual((<any>linearScale)._callbacks.size, 0, "callback was removed from the previous Scale");
+      });
     });
 
-    it("renders correctly given a pixel position", () => {
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let expectedPosition1 = SVG_WIDTH / 2;
-      let gll = new Plottable.Components.GuideLineLayer<number>("horizontal");
-      gll.pixelPosition(expectedPosition1);
-      gll.renderTo(svg);
-
-      assert.strictEqual(gll.content().selectAll(".guide-line").size(), 1, "exactly one line is drawn");
-      let line = gll.content().select(".guide-line");
-      let expectedAttrs1 = {
-        x1: 0,
-        x2: SVG_WIDTH,
-        y1: expectedPosition1,
-        y2: expectedPosition1
-      };
-      TestMethods.assertLineAttrs(line, expectedAttrs1, "the line was drawn at the specified position");
-
-      let expectedPosition2 = SVG_WIDTH * 3 / 4;
-      gll.pixelPosition(expectedPosition2);
-      assert.strictEqual(gll.content().selectAll(".guide-line").size(), 1, "exactly one line is drawn");
-      line = gll.content().select(".guide-line");
-      let expectedAttrs2 = {
-        x1: 0,
-        x2: SVG_WIDTH,
-        y1: expectedPosition2,
-        y2: expectedPosition2
-      };
-      TestMethods.assertLineAttrs(line, expectedAttrs2, "the line was drawn at the updated position");
-
-      svg.remove();
+    describe("Rendering (vertical)", () => {
+      const SVG_WIDTH = 400;
+      const SVG_HEIGHT = 300;
+      const GUIDE_LINE_CLASS = "." + "guide-line";
```

Comment:
Can we look into setting up a TSLint rule for having `const` on variables with ALL_CAPS naming?